addons/mail/static/tests/qunit_suite_tests/components/attachment_link_preview_tests.js

```diff
@@ -0,0 +1,119 @@
+/** @odoo-module **/
+
+import { insert, insertAndReplace } from '@mail/model/model_field_command';
+import {
+    start,
+    startServer,
+} from '@mail/../tests/helpers/test_utils';
+
+QUnit.module('mail', {}, function () {
+QUnit.module('components', {}, function () {
+QUnit.module('attachment_link_preview_tests.js');
+
+QUnit.test('auto layout with link preview', async function(assert) {
+    assert.expect(1);
+
+    const pyEnv = await startServer();
+    const irAttachmentId1 = pyEnv['ir.attachment'].create({
+        description: 'test description',
+        mimetype: 'application/o-linkpreview',
+        name: "https://tenor.com/view/gato-gif-18532922",
+        url: "https://tenor.com/view/gato-gif-18532922",
+    });
+    const mailChannelId1 = pyEnv['mail.channel'].create({});
+    pyEnv['mail.message'].create(
+        {
+            attachment_ids: [irAttachmentId1],
+            model: "mail.channel",
+            res_id: mailChannelId1,
+        }
+    );
+    const { afterEvent, createThreadViewComponent, messaging } = await start();
+    const threadViewer = messaging.models['ThreadViewer'].create({
+        hasThreadView: true,
+        qunitTest: insertAndReplace(),
+        thread: insert({ id: mailChannelId1, model: 'mail.channel' }),
+    });
+    await afterEvent({
+        eventName: 'o-thread-view-hint-processed',
+        func: () => {
+            createThreadViewComponent(threadViewer.threadView);
+        },
+        message: "thread become loaded with messages",
+        predicate: ({ hint, threadViewer }) => {
+            return (
+                hint.type === 'messages-loaded' &&
+                threadViewer.thread.model === 'mail.channel' &&
+                threadViewer.thread.id === mailChannelId1
+            );
+        },
+    });
```

Comment:
Please don't use `createThreadViewComponent` and instead use a main component, such as `openDiscuss`.

We need to adapt all tests that use smaller components, because they are fragile to change, and that's one bottleneck to making some cleaning in the code.

So for new tests, avoid smaller components as much as possible.


Same remark for all newly introduced tests in this file.